support-frontend/app/actions/AsyncAuthenticatedBuilder.scala

```diff
@@ -15,7 +15,7 @@ import scala.concurrent.{ExecutionContext, Future}
 // TODO: considering porting this to identity-play-auth.
 // scalastyle:on
 class AsyncAuthenticatedBuilder[U](
-    userinfo: RequestHeader => Future[Option[U]],
+    userinfo: RequestHeader => Future[U],
```

Comment:
No need to have a future of an option; failed future can be used to model no authenticated user.